src/components/pages/Header.jsx

```diff
@@ -0,0 +1,57 @@
+import { Link, useLocation, useNavigate } from "react-router-dom";
+import { useEffect, useState } from "react";
+
+import DesktopNav from "../navbars/DesktopNav";
+import MobNav from "../navbars/MobNav";
+import ProfileNav from "../navbars/ProfileNav";
+
+const Header = () => {
+    const [isSearch, setIsSearch] = useState(true);
+    const [isLoginBtn, setIsLoginBtn] = useState(true);
+    const [isJoinBtn, setIsJoinBtn] = useState(true);
+    
+    const location = useLocation();
+    useEffect(() => {
+        if (location.pathname === "/search") {
+            setIsSearch(false);
+        } else {
+            setIsSearch(true);
+        }
+        
+        if(location.pathname === "/login") {
+            setIsLoginBtn(false);
+        } else {
+            setIsLoginBtn(true);
+        }
+        
+        if(location.pathname === "/register") {
+            setIsJoinBtn(false);
+        } else {
+            setIsJoinBtn(true);
+        }
+    }, [location.pathname]);
+
+    const navigate = useNavigate();
+    const search = () => {
+        navigate("/search");
+    };
+    const login = () => {
+        navigate("/login");
+    };
+    const register = () => {
+        navigate("/register");
+    };
+
+    return (
+        <header className="h-20 w-full flex items-center px-11 justify-between">
+            <Link className="text-black font-spartan font-semibold text-xl sm:text-3xl w-1/3" to="/">TalentStudio</Link>
```

Comment:
Great job abstracting this into there own components! Really cleans things up and it's much more readable!